app/controllers/application_controller.rb

```diff
@@ -5,6 +5,144 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    if !logged_in?
+      erb :index
+    else
+      erb :'tweets/index'
+    end
+  end
+
+  get '/signup' do
+    if session[:id] != nil
+      redirect "/tweets"
+    else
+      erb :signup
+    end
+  end
+
+  post '/signup' do
+    @user = User.new(
+    username: params[:username],
+    email: params[:email],
+    password: params[:password]
+    )
+    if @user.save
+      session[:id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+  get '/login' do
+    if !logged_in?
+      erb :login
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:id] = @user.id
+      redirect "/tweets"
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect '/login'
+    else
+      redirect '/'
+    end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      erb :'tweets/index'
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/new'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/new' do
+    if params[:content].length.between?(1,141)
```

Comment:
Convention for creating a new tweet would be a post to `/tweets`

```ruby
post '/tweets' do

end
```

You also might want to check to see that the content field is not empty before creating a tweet.